tests/openvino/tools/calibrate.py

```diff
@@ -615,18 +635,43 @@ def get_accuracy_checker_config(engine_config):
 
 
 def get_nncf_algorithms_config(compression_config):
-    nncf_algorithms = []
+    nncf_algorithms = {}
+    override_options = {}
     for pot_algo in compression_config.algorithms:
-        if pot_algo.name not in MAP_POT_NNCF_ALGORITHMS:
-            raise ValueError(f"Algorithm {pot_algo.name} is not supported.")
+        pot_algo_name = pot_algo.name
+        if pot_algo_name not in MAP_POT_NNCF_ALGORITHMS:
+            raise ValueError(f"Algorithm {pot_algo_name} is not supported.")
+
+        nncf_algo_name = MAP_POT_NNCF_ALGORITHMS[pot_algo_name]["method"]
+        nncf_advanced_algo_parameters = MAP_POT_NNCF_ALGORITHMS[pot_algo_name].get("advanced_parameters", {})
+
+        if pot_algo_name in OVERRIDE_OPTIONS_ALGORITHMS:
+            if nncf_algo_name not in override_options:
+                override_options[nncf_algo_name] = defaultdict(dict)
+            parameters = MAP_POT_NNCF_ALGORITHMS[pot_algo_name].get("parameters", {})
+            for p_name, p_value in nncf_advanced_algo_parameters.items():
+                override_options[nncf_algo_name]["advanced_parameters"][p_name] = p_value
+            for p_name, p_value in parameters.items():
+                override_options[nncf_algo_name]["parameters"][p_name] = p_value
+            continue
 
-        nncf_algo_name = MAP_POT_NNCF_ALGORITHMS[pot_algo.name]
-        nncf_algorithms.append(
-            {"name": nncf_algo_name, "parameters": map_paramaters(pot_algo.name, nncf_algo_name, pot_algo.params)}
-        )
+        nncf_algo_parameters = map_paramaters(pot_algo_name, nncf_algo_name, pot_algo.params)
+
+        update_advanced_algo_parameters(nncf_algo_parameters, nncf_advanced_algo_parameters)
```

Comment:
I suggest you try to use the `dataclasses.replace()` method

Reply:
Very useful, thanks. Updated.